elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/MetaDataStore.java

```diff
@@ -80,12 +84,29 @@ public HashMapDataStore apply(String key) {
     private Map<String, HashMapDataStore> hashMapDataStores = new HashMap<>();
 
     public MetaDataStore() {
-        this(ClassScanner.getAnnotatedClasses(METADATA_STORE_ANNOTATIONS));
+        this(getAllAnnotatedClasses());
+    }
+
+    /**
+     * get all MetaDataStore supported annotated classes.
+     * @return Set of Class with specific annotations.
+     */
+    private static Set<Class<?>> getAllAnnotatedClasses() {
+        Set<Class<?>> metadataStoreResult = ClassScanner.getAnnotatedClasses(METADATA_STORE_ANNOTATIONS);
+
+        Set<Class<?>> entityResult = ClassScanner.getAnnotatedClasses(METADATA_STORE_ENTITY_ANNOTATION);
```

Comment:
Class scanning is really time consuming.  We don't want to scan 100% of the classes three times to do this.  In the past, this has caused build times to go through the roof.  We should make maybe a new method in ClassScanner that looks for the classes we need in one pass.